com.unity.uiwidgets/Runtime/gestures/pointer_router.cs

```diff
@@ -56,21 +62,36 @@ public class PointerRouter {
         }
 
         public void route(PointerEvent evt) {
-            HashSet<PointerRoute> routes;
+            // TODO: update this to latest version
+            Dictionary<PointerRoute, Matrix4> routes;
             this._routeMap.TryGetValue(evt.pointer, out routes);
+
+            Dictionary<PointerRoute, Matrix4> copiedGlobalRoutes = new Dictionary<PointerRoute, Matrix4>(this._globalRoutes);
```

Comment:
can we use a shared dictionary instance instead of create a new dictionary each time when calling the function. Since this function might be called very frequently (when user touches here and there), we should take care of the potential gc costs here.

one more question (maybe stupid, just for curiosity), why do we need to copy the global routes? any bad thing would happen if we use the original one?